llvm/lib/TargetParser/Triple.cpp

```diff
@@ -1149,6 +1149,13 @@ std::string Triple::normalize(StringRef Str) {
     }
   }
 
+  // For 3-component triples, the middle component is used to set Vendor;
+  // while if it is "none", we'd prefer to set OS.
+  // This is for some baremetal cases, such as "arm-none-elf".
+  if (Found[0] && !Found[1] && !Found[2] && Found[3] &&
+      Components[1].equals("none") && Components[2].empty())
+    std::swap(Components[1], Components[2]);
```

Comment:
This change, along with the change in libcxx/utils/ci/run-buildbot, look like the only functional changes in this patch. Is it worth committing the non-functional changes (i.e. most of the tests) in a separate NFC patch?

Reply:
Sure. we can. I think that we may need more update to `Triple::normalize`. Once we can decide how to normalize `aarch64-none-elf`, then I will split this PR.